eng/generators.targets

```diff
@@ -1,51 +1,56 @@
 <Project>
 
   <PropertyGroup>
+    <!-- Enable LibraryImportGenerator for CoreLib. -->
     <EnableLibraryImportGenerator Condition="'$(EnableLibraryImportGenerator)' == '' and
                                              '$(MSBuildProjectName)' == 'System.Private.CoreLib'">true</EnableLibraryImportGenerator>
   </PropertyGroup>
 
   <ItemGroup>
     <EnabledGenerators Include="LibraryImportGenerator" Condition="'$(EnableLibraryImportGenerator)' == 'true'" />
-    <!-- If the current project is not System.Private.CoreLib, we enable the LibraryImportGenerator source generator
+    <!-- We enable the Downlevel LibraryImportGenerator
          when the project is a C# source or test project that:
-         - doesn't target the a TFM that includes LibraryImportGenerator or
-         - doesn't reference the live targeting pack (i.e. when inbox) and
-           - references System.Private.CoreLib, or
-           - references System.Runtime.InteropServices -->
-    <EnabledGenerators Include="LibraryImportGenerator"
+         - doesn't target the a TFM that includes LibraryImportGenerator -->
+    <EnabledGenerators Include="DownlevelLibraryImportGenerator"
                        Condition="'$(EnableLibraryImportGenerator)' == '' and
                                   (
                                     '$(IsSourceProject)' == 'true' or
                                     '$(IsTestProject)' == 'true' or
                                     '$(IsTestSupportProject)' == 'true'
                                   ) and
                                   '$(MSBuildProjectExtension)' == '.csproj' and
+                                  !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', '$(NetCoreAppMinimum)'))" />
```

Comment:
This should just be `$(TargetFrameworkIdentifier)' != '.NETCoreApp'` as we shouldn't  target an out-of-support .NETCoreApp TFM.